components/esp_modem/src/esp_modem_dce.cpp

```diff
@@ -191,4 +262,114 @@ modem_mode DCE_Mode::get()
     return mode;
 }
 
+modem_mode DCE_Mode::guess(DTE *dte, bool with_cmux)
+{
+    Scoped<DTE> lock(*dte);
+    return guess_unsafe(dte, with_cmux);
+}
+
+/**
+ * This namespace contains probe packets and expected replies on 3 different protocols,
+ * the modem device could use (as well as timeouts and mode ids for synchronisation)
+ */
+namespace probe {
+
+namespace ppp {
+// Test that we're in the PPP mode by sending an LCP protocol echo request and expecting LCP echo reply
+const uint8_t lcp_echo_request[] = {0x7e, 0xff, 0x03, 0xc0, 0x21, 0x09, 0x01, 0x00, 0x08, 0x99, 0xd1, 0x35, 0xc1, 0x8e, 0x2c, 0x7e };
```

Comment:
Nitpick: This could be a constexpr std::array.

Reply:
used std::array